nautobot_netbox_importer/diffsync/models/abstract.py

```diff
@@ -173,24 +173,36 @@ def clean_ids_or_attrs(  # pylint: disable=too-many-locals
         return (diffsync_data, nautobot_data)
 
     @classmethod
-    def clean_ids(cls, diffsync: DiffSync, ids):
+    def clean_ids(cls, diffsync: DiffSync, ids: dict) -> Tuple[dict, dict]:
         """Translate any DiffSync "ids" fields to the corresponding Nautobot data model identifiers."""
         return cls.clean_ids_or_attrs(diffsync, ids)
 
     @classmethod
-    def clean_attrs(cls, diffsync: DiffSync, attrs):
+    def clean_attrs(cls, diffsync: DiffSync, attrs: dict) -> Tuple[dict, dict]:
         """Translate any DiffSync "attrs" fields to the corresponding Nautobot data model fields."""
         return cls.clean_ids_or_attrs(diffsync, attrs)
 
     @staticmethod
     def create_nautobot_record(nautobot_model, ids: Mapping, attrs: Mapping, multivalue_attrs: Mapping):
         """Helper method to create() - actually populate Nautobot data."""
         try:
+            # Custom fields are a special case - because in NetBox the values defined on a particular record are
+            # only loosely coupled to the CustomField definition itself, it's quite possible that these two may be
+            # out of sync for various reasons. If this happens, Nautobot *will* reject the record when we call clean().
+            # Rather than add a lot of complex logic here to try to "fix" out-of-sync NetBox data, we cheat and
+            # only set the custom field data *after* the record has been validated and approved by Nautobot.
+            custom_field_data = None
+            if "custom_field_data" in attrs:
+                custom_field_data = attrs["custom_field_data"]
+                del attrs["custom_field_data"]
```

Comment:
I know this is a minor tweak, but if this is being called often it could result in a small performance gain by reducing three calls down to one:

```suggestion
            custom_field_data = attrs.pop("custom_field_data", None)
```

Reply:
Every little bit helps - thanks!